packages/report-server/src/reportBuilder/transform/functions/updateColumns.ts

```diff
@@ -28,33 +29,54 @@ export const paramsValidator = yup.object().shape({
   where: yup.string(),
 });
 
-const updateColumns = (rows: Row[], params: UpdateColumnsParams, context: Context): Row[] => {
-  const parser = new TransformParser(rows, context);
-  return rows.map(row => {
-    const returnNewRow = params.where(parser);
-    if (!returnNewRow) {
+const updateColumns = (table: TransformTable, params: UpdateColumnsParams, context: Context) => {
+  const parser = new TransformParser(table, context);
+  const newColumns: Record<string, FieldValue[]> = {};
+  const skippedRows: Record<number, Row> = {};
+  const columnNames = table.getColumns();
```

Comment:
Suggest to clone table first, so the variable `table` is immuable.

Reply:
`getColumns` actually returns a new array of the column names. So mutating those columns won't mutate the original table 👍